### ISSUE TITLE
PermissionError: [Errno 13] Permission denied: '/home/username/tileconvertor/nona'

### ISSUE THREAD

**Issue by GulshanJhaWSP** [mpetroff/pannellum#1185]
I am trying to make a bundle of all the packges required to run generate.py. 
when running and giving the path of nona, it is giving this error. Although I am runnig this command using sudo as 

`sudo python3 generate.py -n /home/username/tileconvertor/nona test.jpg`

any workaround will be very helpul.

**Comment by mpetroff** [comment]
Do you have the executable bit set on `nona`? If you don't, `chmod +x /home/username/tileconvertor/nona` should fix it. I also would advise against running it with `sudo`.

**Comment by GulshanJhaWSP** [comment]
Thank you for the response. 
The main intended to do so is that, I want to integrate this generate.py with another tool (where I can get multires images and put them in a storage and fetch later using panellum). Do I have to deploy that docker container? I tried creating a flask api and attached it with generate.py and creating another docker container.

any suggestions will be very helpful. Thank you :)

**Comment by mpetroff** [comment]
Did setting the execute bit fix the issue or not?

> Do I have to deploy that docker container?

What Docker container?

**Comment by GulshanJhaWSP** [comment]
Hi @mpetroff ,

No, that didn't work after running chmod command. So, Now I have created a docker image with generate.py  (using the docker file in utils/multires). Also, I am getting the desired output tiles and faces of my image.

But, in my level folders, tile size in each level is 512 x 512 except level 1. there I am getting my tiles as 488x488 . Is there something I am doing wrong with the arguments? I have only set tilesize to 512.

Thank You for this amazing tool :)

**Comment by mpetroff** [comment]
> Is there something I am doing wrong with the arguments?

No, that's the expected behavior. Starting from the full cube face resolution, the dimensions of each smaller level is cut in half again and again until the resolution is less than or equal to the tile size. Thus, the level 1 resolution is only equal to the tile size if the full cube face resolution is a power of two; otherwise, it's smaller.

**Comment by GulshanJhaWSP** [comment]
Okay, Got It. 
Thank You.
